File: backend/MeditestApi/Models/Medicine.cs
namespace MeditestApi.Models;

public class Medicine
{
    public Guid Id { get; set; } = Guid.NewGuid();

    /// <summary>Full name of the medicine.</summary>
    public string FullName { get; set; } = string.Empty;

    public string Notes { get; set; } = string.Empty;

    public DateTime ExpiryDate { get; set; }

    public int Quantity { get; set; }

    /// <summary>Price, stored with 2 decimal places.</summary>
    public decimal Price { get; set; }

    public string Brand { get; set; } = string.Empty;
}

File: frontend/meditest-spa/src/app/models/medicine.ts
export interface Medicine {
  id: string;
  fullName: string;
  notes: string;
  expiryDate: string; // ISO date string
  quantity: number;
  price: number;
  brand: string;
}

// Payload for creating a medicine - id is generated server-side.
export type NewMedicine = Omit<Medicine, 'id'>;
